sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1548,6 +1690,167 @@ different from the one with which the dynamic_parameter was created.
 
 |===
 
+=== Event Class Modifications [[event-class-modifications]]
+
+[source, c++]
+----
+namespace sycl::info {
+  enum class ext_oneapi_event_t{
+    normal,
+    graph_limited,
+    graph_external
+  };
+} // namespace sycl::info
+----
+
+:event-class: https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:interface.event
+
+==== Event Type Query
+
+This extension adds the following new information descriptor to the event class
+for use with `event::get_info<>()` which can be used to query the type of a SYCL
+event.
+
+Table {counter: tableNumber}. Additions to the `sycl::event` class information descriptors.
+[%header,cols="2a,a,a"]
+|===
+|Event Descriptor
+|Return Type
+|Description
+
+|`info::event::ext_oneapi_event_type`
+|`info::ext_oneapi_event_t`
+|Returns the type of this SYCL event.
+|===
+
+Events of type `ext_oneapi_event_t::normal` are normal SYCL events as described
+in the SYCL specification. `ext_oneapi_event_t::graph_limited` and
+`ext_oneapi_event_t::graph_external` events are new as part of this extension,
+and are differentiated from normal SYCL events in how they can be used.
+
+==== Graph-Limited Events
+
+Events associated with a graph node (those returned from recorded queue
+submissions) cannot be used as normal SYCL events outside of their associated
+graph, these are known as "graph-limited" events. They can only be used as
+parameters to `handler::depends_on()`, or as dependent events for queue
+shortcuts like `queue::parallel_for()` for other submissions recorded to
+the same modifiable `command_graph`.
+
+The following limitations apply to graph-limited events:
+
+- Calling `event::get_info<info::event::command_execution_status>()` or
+`event::get_profiling_info()` will throw synchronously with error code
+`invalid`.
+
+- A host-side wait on the event will throw synchronously with error
+code `invalid`.
+
+- Using the event outside of the recording scope will throw synchronously with
+error code `invalid`.
+
+==== External Events
+
+To overcome the limitations of "graph-limited" events, users can add external
+event nodes to a graph. These represent a point of execution within the graph
+and provide a SYCL event object which can be used to both manage synchronization
+between different graphs as well as between the graph and regular operations
```

Comment:
regarding "between the graph and regular operations", 

For the case that, 
a) warmup stage:  some regular operations generate an event (name it as eventA).
b) recording stage: it depends on eventA (it is not allowed)

and at the time of warmup stage, the implementation does not know if there will be recording stage.

Any suggestion on how external event can resolve the above issue?

Reply:
I think we could allow this behaviour you suggest, and I have gone back and forth on it a few times. Probably allowing normal SYCL events to be passed to `command_graph::add_wait_external_event()`  and allowing them as dependencies for recorded queue submissions would be fine. 

There may be a potential for some undesirable overhead though: the event dependency in the backend could not be removed from subsequent graph executions once it has completed, so there could be a small overhead there particularly if a lot of these types of events were used (assuming there is still some small overhead from checking completed events in the backend/driver).

The user could avoid this by waiting on those one time operations before executing the graph, but this is problematic in the library record case.

Reply:
I've removed the restrictions on this in the spec for now, and added some wording explicitly allowing it.